mtags/src/main/scala-3/scala/meta/internal/pc/PcDefinitionProvider.scala

```diff
@@ -0,0 +1,202 @@
+  assert(true)
+
+  private def location(p: SourcePosition): Option[Location] = {
+    for {
+      uri <- InteractiveDriver.toUriOption(p.source)
+      range <- if (p.exists) Some(p.toLSP) else None
+    } yield new Location(uri.toString, range)
+  }
+
+  @tailrec
+  private def enclosingSymbols(
+      path: List[Tree],
+      pos: SourcePosition
+  )(using ctx: Context): List[Symbol] = {
+    path match {
+      // For a named arg, find the target `DefDef` and jump to the param
+      case NamedArg(name, _) :: Apply(fn, _) :: _ =>
+        val funSym = fn.symbol
+        if (
+          funSym.name == StdNames.nme.copy
+          && funSym.is(Synthetic)
+          && funSym.owner.is(CaseClass)
+        ) List(funSym.owner.info.member(name).symbol)
+        else {
+          val classTree = funSym.topLevelClass.asClass.rootTree
+          val paramSymbol =
+            for {
+              DefDef(_, paramss, _, _) <- tpd
+                .defPath(funSym, classTree)
+                .lastOption
+              param <- paramss.flatten.find(_.name == name)
+            } yield param.symbol
+          List(paramSymbol.getOrElse(fn.symbol))
+        }
+      case (_: untpd.ImportSelector) :: (imp: Import) :: _ =>
+        importedSymbols(imp, _.span.contains(pos.span))
+
+      case (imp: Import) :: _ =>
+        importedSymbols(imp, _.span.contains(pos.span))
+
+      // For constructor calls, return the `<init>` that was selected
+      case _ :: (_: New) :: (select: Select) :: _ =>
+        List(select.symbol)
+
+      // wildcard param
+      case head :: _ if (head.symbol.is(Param) && head.symbol.is(Synthetic)) =>
+        Nil
+
+      case head :: tl =>
+        if (head.symbol.is(Synthetic)) enclosingSymbols(tl, pos)
+        else if (head.symbol != NoSymbol) List(head.symbol)
+        else {
+          val recovered = recoverError(head)
+          if (recovered.isEmpty) enclosingSymbols(tl, pos)
+          else recovered
+        }
+      case Nil => Nil
+    }
+  }
+
+  private def recoverError(tree: Tree)(using ctx: Context): List[Symbol] = {
+    val recoveredDenot = tree match {
+      case select: Select => Some(select.qualifier.typeOpt.member(select.name))
+      case ident: Ident =>
+        val ref = ctx.typer.findRef(
+          ident.name,
+          dotty.tools.dotc.core.Symbols.defn.AnyType,
+          EmptyFlags,
+          EmptyFlags,
+          NoSourcePosition
+        )
+        ref match {
+          case v: NamedType => Some(v.denot)
+          case _ => None
+        }
+      case _ => None
+    }
+
+    recoveredDenot match {
+      case Some(multi: MultiPreDenotation) =>
+        List(multi.last, multi.first)
+          .map(_.symbol)
+          .filter(_ != NoSymbol)
+      case Some(denot) => List(denot.symbol).filter(_ != NoSymbol)
+      case None => List.empty
+    }
+  }
+
+  def symOrdering(using Context): Ordering[Symbol] =
+    new Ordering[Symbol] {
+      def compare(first: Symbol, second: Symbol): Int = {
```

Comment:
We might compare based on semanticdb symbols, I think that would influence the `assert` ordering. Or alternatively use the one in Completions, where I added the method signature.

Reply:
Completions doesn't fit for that. Changed this part a little bit